### ISSUE TITLE
Sub-header: TypeError: e[0].querySelector is not a function

### ISSUE THREAD

**Issue by alanpurple** [angular/material#3813]
h2 class="md-subheader ng-scope md-default-theme" ng-if="car.trims"

error occured in angular.js by angular-material.js

**Comment by poltorak** [comment]
+1

it happens because there's attempt to get querySelector on ```comment``` element 

https://github.com/angular/material/blob/master/src/components/subheader/subheader.js#L60

**Comment by topherfangio** [comment]
I believe this was fixed in PR #4171. Can someone confirm if you are still seeing this in master?

I just added couple of test cases (locally only) to `subheader.spec.js` which appear to pass. If they are helpful, I can submit a PR to show the functionality works.

```js

  it('supports ng-if = true', function() {
    build('<div><md-subheader ng-if="true">test</md-subheader></div>');

    expect($exceptionHandler.errors).toEqual([]);
    expect(element[0].querySelectorAll('.md-subheader').length).toEqual(1);
  });

  it('supports ng-if = false', function() {
    build('<div><md-subheader ng-if="false">test</md-subheader></div>');

    expect($exceptionHandler.errors).toEqual([]);
    expect(element[0].querySelectorAll('.md-subheader').length).toEqual(0);
  });

  it('supports ng-if dynamically', function() {
    build('<div><md-subheader ng-if="showHeader">test</md-subheader></div>');

    pageScope.$apply('showHeader = true');

    expect($exceptionHandler.errors).toEqual([]);
    expect(element[0].querySelectorAll('.md-subheader').length).toEqual(1);
  });
```

**Comment by ThomasBurleson** [comment]
Closing for now.
